### PR TITLE
Add test for victory legend groupComponent

### PR DESCRIPTION
**Pull request by jeffreypriebe**
This test fails on v0.21.3; passes on v0.21.2
See [Issue #692 in Victory repo](https://github.com/FormidableLabs/victory/issues/692).

When passing a `groupComponent` to Victory Legend, in 0.21.2 the component would get valid values for `width` and `height`. In 0.21.3 both `width` and `height` are 0.

It looks like this bug was introduced in 7432a72 where previously the victory-legend `render` method would call `getCalculatedProps` where the width & height would then get populated.
Similar code is now in the 'victory-legend/helper-methods.js': `calculateLegendHeight` and `calculateLegendWidth` which are called from the exported `default`. However, that default is imported in 'victory-legend' as `getBaseProps` and then *not called*. (In render, [the props are constructed](https://github.com/FormidableLabs/victory-core/commit/7432a7292febca88e7e18800b462c0d49d26f00d#diff-6da5f8e213cd6fa075b7f42184b3da09R100) - not calling `getBaseProps` and the `renderContainer` isn't actually passed those constructed props.)

I tried to fix this by calling `getBaseProps` but was unable to figure out how that should be done inside the `VictoryLegend` component.

### CODE REVIEW

**Comment by boygirl** [comment]
The change from `0.21.2` to `0.21.3` enforced consistency between `VictoryLegend` and other Victory components.

The props passed to `containerComponent` are specifically excluded from `groupComponent`. Most passed props cause a warning (unexpected prop x supplied to g... ) Though `width` and `height` don't cause this error, they also don't have any effect because,  svg `<g>` elements don't have explicit height and width attributes, they auto size to whatever they contain.

So, this test _should_ fail. Can you tell me more about your use case?

(p.s. If you're writing a custom `groupComponent`, and you'd like access to the props that are usually passed to your `containerComponent` you can add `static role = "container";` to your `groupComponent`)

**Comment by jeffreypriebe** [comment]
Thanks for the response here and also for working on this open-source library.

1. Is there a new way to create groups/containers in `0.21.3` that I should be using?
2. You say "The props passed to `containerComponent` are specifically excluded from `groupComponent`". However, as mentioned, those props are being passed - with a value of '0'.
3. The `default` and `calculateLegendHeight` etc. in 'victory-legend/helper-methods.js' are not being used since https://github.com/FormidableLabs/victory-core/commit/7432a7292febca88e7e18800b462c0d49d26f00d, is that the intended outcome?
The code was moved to 'helper-methods' and now isn't being called.

#### My Use Case
My use case is quite simple: wrapping the legend and drawing a `<rect />` with a `stroke` value and then positioning it in the top-right of the chart.  Having the calculated width of the legend contents allows me to position the legend top right as well as draw the `<rect />` of the correct dimensions.

Basically, my `groupComponent` is essentially (simplified):
```
const { width, height } = this.props;
const translateX = chartWidth - width - inset;
return (
  <g transform={`translate(${translateX}, 0)`}>
    <rect width={width + padding} height={height + padding} stroke="#000" strokeWidth="1" />
    {children}
  </g>
);
```

**Comment by boygirl** [comment]
@jeffreypriebe I think you can get your custom group component working by adding `static role = "container";` Let me think about how to make this more straightforward...

**Comment by jeffreypriebe** [comment]
@boygirl Would the `role='container'` somehow provide the `width` & `height` values?

**Comment by jeffreypriebe** [comment]
@boygirl As you think about it, I've tested and `static role = 'container'` still has no width or height values.

The `getCalculatedProps` still isn't being called.

Are you open to using `getCalculatedProps` if I can figure out how it should be called? Or is there is a reason from 7432a72 that these values are now unwanted?

**Comment by boygirl** [comment]
@jeffreypriebe sorry I missed your last note. The `width` and `height` values _are_ provided to your container.

They're calculated here: https://github.com/FormidableLabs/victory-core/blob/master/src/victory-legend/helper-methods.js#L82

and then added to base parent props here: https://github.com/FormidableLabs/victory-core/blob/master/src/victory-legend/helper-methods.js#L85

Then everything in on base parent props is sent to the containerComponent (when static role = container). You can see that logic here: https://github.com/FormidableLabs/victory-core/blob/master/src/victory-util/add-events.js#L152

All that said. I think providing a border around a legend is a common enough use case that I should make a much easier way to accomplish it. I will open a new issue.
